### PR TITLE
add strain

### PR DESCRIPTION
**Pull request by BNAndras**

### CODE REVIEW

File: config.json
Comment (glennj) [inline config.json:1]: The same functionality for this exercise also shows up in `list-ops`. Can you please make sure that `strain` appears earlier than list-ops, with a lower difficulty?

File: exercises/practice/strain/.meta/proof.ci.wren [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    for (item in list) {
+      if (predicate.call(item)) {
+        kept.add(item)
+      }
+    }
+    return kept
+  }
+
+  static discard(list, predicate) {
+    return keep(list, Fn.new { |x| !predicate.call(x) })
```
Comment (glennj) [inline exercises/practice/strain/.meta/proof.ci.wren:13]: Perhaps?
```suggestion
    return keep(list) { |x| !predicate.call(x) }
```

File: exercises/practice/strain/strain.spec.wren [hunk 2]
```diff
@@ -0,0 +1,112 @@
+import "wren-testie/testie" for Testie, Expect
+import "./strain" for Strain
+
+Testie.test("Strain") { |do, skip|
+  do.describe("keep") {
+    do.test("keep on empty list returns empty list") {
+      var predicate = Fn.new { |x| true }
+      Expect.value(Strain.keep([], predicate)).toEqual([])
```
Comment (glennj) [inline exercises/practice/strain/strain.spec.wren:8]: more idiomatic to pass the predicate as a block (https://wren.io/functions.html#block-arguments):
```suggestion
      var actual = Strain.keep([]) { |x| true }
      Expect.value(actual).toEqual([])
```

**Review by glennj — CHANGES_REQUESTED** [review]

**Comment by glennj** [comment]
This might be an opportunity to add an instructions.append.md file pointing students to [the Fn class](https://wren.io/modules/core/fn.html#call(args...)) and the `call` method.

File: config.json
Comment (BNAndras) [reply config.json:1]: I'll take a look tomorrow. Thanks for the review

**Comment by BNAndras** [comment]
Updated

**Review by glennj — APPROVED** [review]

**Comment by glennj** [comment]
Very nice, thank you.

**Pull request merged**